tests/e2e/Services/Functions/FunctionsCustomClientTest.php

```diff
@@ -963,4 +963,52 @@ public function testNonOverrideOfHeaders(): array
 
         return [];
     }
+
+    public function testGetFunctionTemplates()
+    {
+        $templates = $this->client->call(Client::METHOD_GET, '/functions/templates', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $templates['headers']['status-code']);
+        $this->assertGreaterThan(0, $templates['body']['total']);
+        $this->assertIsArray($templates['body']['templates']);
+        $this->assertArrayHasKey('runtimes', $templates['body']['templates'][0]);
+
+        $templates = $this->client->call(Client::METHOD_GET, '/functions/templates', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'usecases' => ['starter', 'ai'],
+            'runtimes' => ['bun-1.0', 'dart-2.16']
+        ]);
+
+        $this->assertEquals(200, $templates['headers']['status-code']);
+        $this->assertEquals(3, $templates['body']['total']);
+        $this->assertIsArray($templates['body']['templates']);
+        $this->assertArrayHasKey('runtimes', $templates['body']['templates'][0]);
+        $this->assertEquals('starter', $templates['body']['templates'][0]['usecases'][0]);
+        $this->assertEquals('ai', $templates['body']['templates'][1]['usecases'][0]);
+        $this->assertEquals('ai', $templates['body']['templates'][2]['usecases'][0]);
+        $this->assertContains('bun-1.0', array_column($templates['body']['templates'][0]['runtimes'], 'name'));
+        $this->assertContains('dart-2.16', array_column($templates['body']['templates'][1]['runtimes'], 'name'));
+
+        $templates = $this->client->call(Client::METHOD_GET, '/functions/templates', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'limit' => 5,
+            'offset' => 2,
```

Comment:
Offset isnt assteted in any way. We need to load with offset 0, and then with offset 2. And ensure first 2 are missing, but third becomes first.